proxy/tcpproxy/userspace.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tcpproxy
+
+import (
+	"io"
+	"net"
+	"sync"
+	"time"
+)
+
+type tcpProxy struct {
+	l               net.Listener
+	monitorInterval time.Duration
+
+	mu      sync.Mutex // guards the following fields
+	remotes []*remote
+	next    int
+}
+
+type remote struct {
+	mu       sync.Mutex
+	addr     string
+	inactive bool
```

Comment:
the problem is the initial state should be active. but the default value of bool is false. so we need to be more careful about initializing remote. i do not have strong opinion on this.